plugins/lang-plan-a/src/main/java/org/elasticsearch/plan/a/Compiler.java

```diff
@@ -30,125 +30,134 @@
-        final ParserErrorStrategy strategy = new ParserErrorStrategy();
-
-        lexer.removeErrorListeners();
-        lexer.setTypes(definition.structs.keySet());
-
-        //List<? extends Token> tokens = lexer.getAllTokens();
-
-        //for (final Token token : tokens) {
-        //    System.out.println(token.getType() + " " + token.getText());
-        //}
+    /**
+     * Runs the two-pass compiler to generate a Plan A script.
+     * @param loader The ClassLoader used to define the script.
+     * @param name The name of the script.
+     * @param source The source code for the script.
+     * @param settings The CompilerSettings to be used during the compilation.
+     * @return An {@link Executable} Plan A script.
+     */
+    static Executable compile(final Loader loader, final String name, final String source,
+                              final Definition custom, final CompilerSettings settings) {
+        // Set the definition.
+        final Definition definition = custom != null ? new Definition(custom) : DEFAULT_DEFINITION;
 
-        parser.removeErrorListeners();
-        parser.setErrorHandler(strategy);
-
-        ParserRuleContext root = parser.source();
-
-        //end = System.currentTimeMillis() - start;
-        //System.out.println("tree: " + end);
-
-        final Adapter adapter = new Adapter(definition, source, root, settings);
-
-        start = System.currentTimeMillis();
+        // Generate the ANTLR tree from the given source.
+        final ParserRuleContext root = createParseTree(source, definition);
 
-        Analyzer.analyze(adapter);
-        //System.out.println(root.toStringTree(parser));
+        // Create the Metadata (wrapper) used for this script.
+        final Metadata metadata = new Metadata(definition, source, root, settings);
 
-        //end = System.currentTimeMillis() - start;
-        //System.out.println("analyze: " + end);
-        //start = System.currentTimeMillis();
+        // Run the Analyzer to generate the metadata necessary for writing the byte code.
+        Analyzer.analyze(metadata);
 
-        final byte[] bytes = Writer.write(adapter);
-
-        //end = System.currentTimeMillis() - start;
-        //System.out.println("write: " + end);
-        //start = System.currentTimeMillis();
+        // Run the Writer to generate the byte code using ASM.
+        final byte[] bytes = Writer.write(metadata);
 
+        // Create the executable from the given ClassLoader.
         final Executable executable = createExecutable(loader, definition, name, source, bytes);
 
-        //end = System.currentTimeMillis() - start;
-        //System.out.println("create: " + end);
-
         return executable;
     }
 
-    private static ParserRuleContext createParseTree(String source, Definition definition) {
+    // Generates the ANTLR tree from the given source code.  Several methods, as noted below, are used
```

Comment:
I prefer javadoc even for these private methods because I'm used to it. I'd probably use a block comment instead of line comments even if I didn't use javadoc just because my code formatter does a better job on block comments that take up multiple lines than multiple contiguous line comments like this. You can do what you like. We have all styles in 
Elasticsearch.